timemachine/fe/mcgregor.py

```diff
@@ -244,13 +167,191 @@ def to_networkx(self) -> nx.Graph:
         return g
 
 
-def max_tree_size(priority_list):
-    cur_layer_size = 1
-    layer_sizes = [cur_layer_size]
-    for neighbors in priority_list:
-        cur_layer_size *= len(neighbors)
-        layer_sizes.append(cur_layer_size)
-    return sum(layer_sizes)
+@dataclass(frozen=True)
+class Marcs:
+    marcs: NDArray[np.bool_]
+
+    @cached_property
+    def num_edges_upper_bound(self) -> int:
```

Comment:
Added a note in https://github.com/proteneer/timemachine/pull/1415/commits/db8391e13a4e014d481a75eea5725e750cddbb89. Personally I find "arcs left" to be confusing since it somehow suggests to me that it does not include the edges that are already in correspondence at a given point in the search, only the ones "left" to possibly map. (I think it's just using "left" in a different sense that I first assumed.)